sql/hive/src/main/scala/org/apache/spark/sql/hive/hiveUdfs.scala

```diff
@@ -179,7 +179,12 @@ private[hive] case class HiveGenericUdf(funcWrapper: HiveFunctionWrapper, childr
         })
       i += 1
     }
-    unwrap(function.evaluate(deferedObjects), returnInspector)
+    
+    if (function.getUdfName().endsWith("UDFCurrentDB")) {
```

Comment:
Yes. Because the name of currentDB has been contained in returnInspector when init, and  the name will be getted from returnInspector  firstly in function "unwrap",So it is unnecessary that get the name again.